libs/components/indicators/src/lib/modules/text-highlight/text-highlight.directive.ts

```diff
@@ -27,160 +114,98 @@ export class SkyTextHighlightDirective
    * Specifies the text to highlight.
    */
   @Input()
-  public set skyHighlight(value: string | string[]) {
+  public set skyHighlight(value: string | string[] | undefined) {
     value = value || [];
-    this._skyHighlight = [];
 
     if (Array.isArray(value)) {
-      this._skyHighlight = value.filter((item) => !!item);
+      this.#skyHighlightValues = value.filter((item) => !!item);
       // Reorder strings by their length in descending order to avoid missing matches
       // that contain substrings of other matches.
-      this._skyHighlight.sort(function (a, b) {
+      this.#skyHighlightValues.sort(function (a, b) {
         return b.length - a.length;
       });
     } else {
-      this._skyHighlight = [value as string];
+      this.#skyHighlightValues = [value as string];
     }
   }
 
-  private existingHighlight = false;
-
-  private observer: MutationObserver;
-
-  private _skyHighlight: string[];
-
-  constructor(
-    private el: ElementRef,
-    private observerService: MutationObserverService
-  ) {}
-
-  private static cleanRegex(regex: string): string {
-    return regex.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&');
-  }
-
-  private static markNode(node: any, searchTerms: string[]): number {
-    /* istanbul ignore else */
-    if (searchTerms) {
-      const text = node.nodeValue;
-      for (let i = 0; i < searchTerms.length; i++) {
-        searchTerms[i] = this.cleanRegex(searchTerms[i]);
```

Comment:
This is where the bug where search terms were escaped multiple times originated. The original array would be updated with a re-escaped version of each item for every child element, so when a certain number of child elements were present, the regular expression reached a size so large it would cause an error. For example, the first iteration would escape `'.'` with `'\.'`. The second iteration would escape the already-escaped string `'\.'` with `'\\\.'`, and so on.